src/index.tsx

```diff
@@ -261,20 +280,22 @@ function UseWalletProvider({
       return
     }
 
-    const { ethereum } = window
-    ethereum.enable.then(() => {
-      if (!ethereum?.selectedAddress) {
-        return
-      }
+    const lastConnector = getLastConnector()
+    const lastActiveAccount = getLastActiveAccount()
 
-      const isInjectedAvailable = Object.keys(connectors).some(
-        (key) => key === 'injected'
-      )
+    const { ethereum } = window
+    if (lastActiveAccount && ethereum && lastConnector === 'injected') {
+      ethereum.enable()?.then(() => {
```

Comment:
the whole `ethereum` stuff should no longer be necessary, right?

Reply:
True... removed